### PR TITLE
Fix Tags; Add Firewall ID, User Data; Other MISC Updates

### LINKED ISSUE #220: [Bug]: Tags are not applied when building using Packer
tags
### Packer Version

v1.10.2

### Packer Plugin Linode Version

v1.4.1

### Expected Behavior

VM tags are not applied to the VM when building using Packer.  

```hcl
source "linode" "example" {
  image             = "linode/ubuntu22.04"
  image_description = "This image was created using Packer."
  image_label       = "test-image-${local.timestamp}"
  instance_label    = "temp-packer-debian-10"
  instance_type     = "g6-nanode-1"
  linode_token      = "${var.linode_api_token}"
  region            = "nl-ams"
  ssh_username      = "root"
  instance_tags     = ["foo"]
 
}
```

### Actual Behavior

Command output: 
```
packer build .
linode.example: output will be in this color.

==> linode.example: Running builder ...
==> linode.example: Creating temporary SSH key for instance...
==> linode.example: Creating Linode...
==> linode.example: Using SSH communicator to connect: 172.233.51.105
==> linode.example: Waiting for SSH to become available...
==> linode.example: Connected to SSH!
==> linode.example: Provisioning with shell script: /tmp/packer-shell3435650164
    linode.example: Adding file to Docker Container
==> linode.example: Shutting down Linode...
==> linode.example: Creating image...
```
Build is done successfully, but tags are not being applied to the VM (screenshot attached)

### Steps to Reproduce

Specify VM tags using "instance_tags"

```hcl
source "linode" "example" {
  image             = "linode/ubuntu22.04"
  image_description = "This image was created using Packer."
  image_label       = "test-image-${local.timestamp}"
  instance_label    = "temp-packer-debian-10"
  instance_type     = "g6-nanode-1"
  linode_token      = "${var.linode_api_token}"
  region            = "nl-ams"
  ssh_username      = "root"
  instance_tags     = ["foo"]
 
}
```

### Error Output

No error output. Tags are not being applied.

### LINKED ISSUE #221: [Bug]: Equal (=) sign is not being accepted as a tag value
### Packer Version

v1.10.2

### Packer Plugin Linode Version

v1.4.1

### Expected Behavior

Tag which contains an equal sign are not being accepted as a valid character. Adding it via the API or web interface works without any issues. 

Looks like the limitation comes from this regex - https://github.com/linode/packer-plugin-linode/blob/5d07271dad0fd6d2f518096e90ba98fd3ee0e619/builder/linode/config.go#L160

```
source "linode" "example" {
  image             = "linode/ubuntu22.04"
  image_description = "This image was created using Packer."
  image_label       = "test-image-${local.timestamp}"
  instance_label    = "temp-packer-debian-10"
  instance_type     = "g6-nanode-1"
  linode_token      = "${var.linode_api_token}"
  region            = "nl-ams"
  ssh_username      = "root"
  instance_tags     = ["foo=bar"]
}
```

### Actual Behavior

Error returned

### Steps to Reproduce

Add a tag which contains = sign

```
source "linode" "example" {
  image             = "linode/ubuntu22.04"
  image_description = "This image was created using Packer."
  image_label       = "test-image-${local.timestamp}"
  instance_label    = "temp-packer-debian-10"
  instance_type     = "g6-nanode-1"
  linode_token      = "${var.linode_api_token}"
  region            = "nl-ams"
  ssh_username      = "root"
  instance_tags     = ["foo=bar"]
}
```

### Error Output

```
Error: 1 error(s) occurred:

* invalid tag: foo=bar

  on linode.pkr.hcl line 1:
  (source code not available)
```

### PR DESCRIPTION
**Pull request by zliang-akamai**
## 📝 Description

Multiple updates bundled:

- [Error checking](https://github.com/linode/packer-plugin-linode/pull/223/files#diff-335f28fa1ddbfd616457fe479eceddd30f11d59e50824abbad472b00f7c6412eL45-L49) that checks an always-nil `err` has been removed.
- Many `interface{}` were replaced with `any`.
- Add support for `firewall_id` and `user_data` of `metadata` attribute.
- `instance_tags` and its validation regular expression have been fixed.

Closes #221
Closes #220

### Automated Testing

```bash
make test
```

### Manual Testing

- Save this in a file called `test.pkr.hcl`.
```hcl
source "linode" "example" {
  image             = "linode/ubuntu22.04"
  instance_type     = "g6-nanode-1"
  region            = "us-mia"
  ssh_username      = "root"
  instance_tags     = ["abc", "foo=bar"]
  firewall_id       = 12345 // use your firewall ID here
  metadata {
    user_data = base64encode(<<EOF
#cloud-config

write_files:
  - path: /root/helloworld.txt
    content: |
      Hello, world!
    owner: 'root:root'
    permissions: '0644'
EOF
)
  }
}

build {
  sources = ["source.linode.example"]
}

```

- `go build`
- `packer build .`

### CODE REVIEW

**Review by lgarber-akamai — APPROVED** [review]
LGTM, thank you for addressing this!

File: builder/linode/config.go [hunk 1]
```diff
@@ -40,24 +44,26 @@ type Config struct {
 	ctx                 interpolate.Context
 	Comm                communicator.Config `mapstructure:",squash"`
 
-	Interfaces         []Interface       `mapstructure:"interface"`
+	Interfaces         []Interface       `mapstructure:"interface" required:"false"`
```
Comment (lgarber-akamai) [inline builder/linode/config.go:47]: Just curious, what does the `required:"false"` here do? Are these fields required by default?

Comment (zliang-akamai) [reply builder/linode/config.go:47]: I believe that it's just a stylistic change to explicitly mark them as optional attributes.

**Review by yec-akamai — APPROVED** [review]
Great work!

**Pull request merged**